### PR TITLE
Create schema.graphql

### PR DESCRIPTION
**Pull request by usm076**

### CODE REVIEW

**Review by sheenhx — COMMENTED** [review]
# Pull Request Review

## Summary

This pull request adds a new file called `schema.graphql` which contains GraphQL schema definitions for various entities related to an NFT marketplace. The schema includes the following types:

1. ChangedOwnerCutPerMillion
2. ChangedPublicationFee
3. MetaTransactionExecuted
4. OrderCancelled
5. OrderCreated
6. OrderSuccessful
7. OwnershipTransferred
8. Paused
9. Unpaused
10. MXCCollectionFactoryV1OwnershipTransferred
11. collection
12. Approval
13. ApprovalForAll
14. MXCNFTTokenV1OwnershipTransferred
15. Transfer
16. mint

Each type has several fields, most of which are either of type `Bytes`, `BigInt`, or `String`. All types have the `@entity(immutable: true)` directive, indicating that they are immutable entities.

## Typo Check

There are no typos found in the schema definitions.

## Logic

The logic of the schema seems to be focused on representing different events and actions that can occur within an NFT marketplace, such as creating orders, canceling orders, transferring ownership, approving tokens for transfer, and minting new tokens.

Overall, the schema appears to be well-structured and covers a wide range of functionality needed for an NFT marketplace application.

## Suggested Changes

No changes are necessary as the schema is well-defined and free of any typos or errors.

---

In conclusion, this pull request adds a comprehensive GraphQL schema for an NFT marketplace application with well-defined types and fields covering various events and actions that can occur within the marketplace. The schema is free of any typos or errors and does not require any changes at this time.

**Review by sheenhx — COMMENTED** [review]
## Summary

This pull request contains two patches:

### Patch 1: Create schema.graphql

This patch creates a new file `schema.graphql` with the following GraphQL entities:

- ChangedOwnerCutPerMillion
- ChangedPublicationFee
- MetaTransactionExecuted
- OrderCancelled
- OrderCreated
- OrderSuccessful
- OwnershipTransferred
- Paused
- Unpaused
- MXCCollectionFactoryV1OwnershipTransferred
- collection
- Approval
- ApprovalForAll
- MXCNFTTokenV1OwnershipTransferred
- Transfer
- mint

These entities represent various events and data structures related to a marketplace for NFTs.

### Patch 2: marketplace and collection factory.ts files

This patch creates two new TypeScript files, `src/marketplace-v-1.ts` and `src/mxc-collection-factory-v-1.ts`, which handle events from the MarketplaceV1 and MXCCollectionFactoryV1 smart contracts respectively.

In `src/marketplace-v-1.ts`, event handlers are defined for the following events:

- ChangedOwnerCutPerMillionEvent
- ChangedPublicationFeeEvent
- MetaTransactionExecutedEvent
- OrderCancelledEvent
- OrderCreatedEvent
- OrderSuccessfulEvent
- OwnershipTransferredEvent
- PausedEvent
- UnpausedEvent

In `src/mxc-collection-factory-v-1.ts`, event handlers are defined for the following events:

- OwnershipTransferredEvent (MXCCollectionFactoryV1)
- collectionEvent

## Review Notes

The code seems well-written and follows best practices. There are no typos or syntax errors. The logic is clear, and the event handlers update the corresponding entities in response to emitted events from the smart contracts.

**Review by sheenhx — COMMENTED** [review]
## Summary

This pull request contains 3 patches that introduce a GraphQL schema, event handlers for the MarketplaceV1, MXCCollectionFactoryV1, and MXCNFTTokenV1 contracts, and a subgraph configuration file.

### Patch 1: Create schema.graphql

This patch introduces the `schema.graphql` file with various entity types. The entities include:

- ChangedOwnerCutPerMillion
- ChangedPublicationFee
- MetaTransactionExecuted
- OrderCancelled
- OrderCreated
- OrderSuccessful
- OwnershipTransferred
- Paused
- Unpaused
- MXCCollectionFactoryV1OwnershipTransferred
- collection
- Approval
- ApprovalForAll
- MXCNFTTokenV1OwnershipTransferred
- Transfer
- mint

### Patch 2: marketplace and collection factory.ts files

This patch adds two TypeScript files:

1. `src/marketplace-v-1.ts`: This file contains event handlers for the MarketplaceV1 contract events.
2. `src/mxc-collection-factory-v-1.ts`: This file contains event handlers for the MXCCollectionFactoryV1 contract events.

### Patch 3: yaml file and mxc nft

This patch adds two more files:

1. `src/mxcnft-token-v-1.ts`: This file contains event handlers for the MXCNFTTokenV1 contract events.
2. `subgraph.yaml`: This is the subgraph configuration file that defines data sources and mappings for the contracts.

## Review Findings

The code appears to be well-written and follows best practices. However, there are some minor issues that need to be addressed:

### Typo

In `src/mxc-collection-factory-v-1.ts`, the function name should be `handleCollection` instead of `handlecollection`.

```diff
-export function handlecollection(event: collectionEvent): void {
+export function handleCollection(event: collectionEvent): void {
```

### Consistency

In `subgraph.yaml`, the event handler names should be consistent with the function names in the corresponding TypeScript files. For example, change `handlecollection` to `handleCollection`.

```diff
-        - event: collection(address,string,address,string)
-          handler: handlecollection
+        - event: collection(address,string,address,string)
+          handler: handleCollection
```

Overall, the pull request looks good with only minor issues that need to be addressed.
